### PR TITLE
Use invisible header row

### PR DESCRIPTION
**Pull request by tmpacifitech**
Fixes https://github.com/plotly/dash-table/issues/205 and https://github.com/plotly/dash-table/issues/432

### CODE REVIEW

**Review by akksi — COMMENTED** [review]
It kind of works, but it's quite flickery/laggy and sometimes unresponsive when scrolling vertically or resizing the window.

https://user-images.githubusercontent.com/14875243/133575525-90e658f5-5eae-4233-b2cc-75580ed8766a.mp4

I'm not sure how much work it would take, but maybe it would be worth restructuring the HTML table markup a bit (by joining the `header` and the `body` part to one `table`) to render a semantic table and format it with CSS instead of calculating the column sizes with js.

If not, maybe it's possible to optimize the calculations a bit more.

**Comment by akksi** [comment]
A lower effort solution could be cloning the header row into the table body and make it visually hidden so it would essentially set the min. width of the columns - in theory.

**Comment by tmpacifitech** [comment]
@akksi can you take a look at this PR again?

**Comment by akksi** [comment]
> @akksi can you take a look at this PR again?

Looks good to me. :)

File: components/dash-table/tests/visual/percy-storybook/DashTable.percy.tsx [hunk 1]
```diff
@@ -165,10 +179,21 @@ storiesOf('DashTable/Fixed Rows & Columns', module)
...
+    .add('with 1 fixed row, longer header width than content cell', () => (
+        <DataTable
+            setProps={setProps}
+            id='table'
+            data={dataWithLongerTitle}
+            columns={columnsWithLongerTitle}
+            fixed_rows={{headers: true}}
+            row_deletable={true}
+            row_selectable='single'
+        />
```
Comment (chriddyp) [inline components/dash-table/tests/visual/percy-storybook/DashTable.percy.tsx:195]: image

Could we add a test where the columns are as wide as `max(header, cell)`? In this test case, the table is very wide so we don't see the issue of the narrow headers.

Comment (chriddyp) [reply components/dash-table/tests/visual/percy-storybook/DashTable.percy.tsx:195]: Perhaps a test with several columns

**Comment by chriddyp** [comment]
This percy diff is odd - Looks like we lost the header border for the row selection column?
image

**Comment by chriddyp** [comment]
Another odd one - custom colors aren't being applied to that border
image

File: components/dash-table/tests/visual/percy-storybook/DashTable.percy.tsx (hunk 1 above)
Comment (chriddyp) [reply components/dash-table/tests/visual/percy-storybook/DashTable.percy.tsx:195]: Ah well I guess this existing test covers it:
image

**Comment by chriddyp** [comment]
Looks like this test is missing data now
`DashTable/Border (available space filled): with frozen rows and no frozen columns`
image

https://percy.io/plotly/dash-table-test/builds/12769932/changed/721646457?browser=chrome&browser_ids=19%2C20%2C21&subcategories=unreviewed%2Cchanges_requested&viewLayout=side-by-side&viewMode=new&width=1280&widths=375%2C1280

**Comment by chriddyp** [comment]
And this one isn't getting rendered

https://percy.io/plotly/dash-table-test/builds/12769932/changed/721646552?browser=firefox&browser_ids=19%2C20%2C21&subcategories=unreviewed%2Cchanges_requested&viewLayout=side-by-side&viewMode=new&width=1280&widths=375%2C1280

image

**Comment by chriddyp** [comment]
The widths of these columns changed considerably. Not sure what the "true" width should be for this case though.

image

**Comment by chriddyp** [comment]
Same here
https://percy.io/plotly/dash-table-test/builds/12769932/changed/721646761?browser=edge&browser_ids=19%2C20%2C21&subcategories=unreviewed%2Cchanges_requested&viewLayout=side-by-side&viewMode=new&width=1280&widths=375%2C1280

image

Again not sure what the width "should" be in this case.

**Comment by chriddyp** [comment]
There is a double border below A now

https://percy.io/plotly/dash-table-test/builds/12769932/changed/721646637?browser=firefox&browser_ids=19%2C20%2C21&subcategories=unreviewed%2Cchanges_requested&viewLayout=side-by-side&viewMode=new&width=1280&widths=375%2C1280

image

**Comment by gvwilson** [comment]
@tmpacifitech we think most of the Dash DataTable has been superseded by AG Grid, so we're going to close this one as stale. Thank you again for having contributed - sorry it took us so long to get to this one.

**Pull request closed without merging**